apps/functions/applications-background-jobs/malware-detected/src/html-validation.js

```diff
@@ -0,0 +1,59 @@
+import { blobClient } from './blob-client.js';
+import config from './config.js';
+import { streamToString } from './util.js';
+import { YouTubeHTMLTemplate } from './youtube-html-expectation.js';
+
+/**
+ * @param {string} blobUri
+ * @returns {Promise<boolean | null>}
+ * */
+export const isScannedFileHtml = async (blobUri) => {
+	const blobProperties = await blobClient.getBlobProperties(config.BLOB_SOURCE_CONTAINER, blobUri);
+	return blobProperties.contentType === 'text/html';
+};
+
+/**
+ * @param {string} blobUri
+ * */
+export const isUploadedHtmlValid = async (blobUri) => {
+	const downloadBlockBlobResponse = await blobClient.downloadStream(
+		config.BLOB_SOURCE_CONTAINER,
+		blobUri
+	);
+
+	if (!downloadBlockBlobResponse.readableStreamBody) {
+		throw Error('No file stream received after attempting to download from Blob Container');
+	}
+	const downloadedContent = await streamToString(downloadBlockBlobResponse.readableStreamBody);
+
+	return compareHtmlStringsIgnoringYoutubeUrl(downloadedContent, YouTubeHTMLTemplate);
+};
+
+/**
+ * @param {String} html1
+ * @param {String} html2
+ * @returns {Boolean}
```

Comment:
Not sure if casing matters for types 🤔 

```suggestion
 * @param {string} html1
 * @param {string} html2
 * @returns {boolean}
```